src/EventHub/EventHub/Cmdlets/Clusters/NewAzureEventHubCluster.cs

```diff
@@ -0,0 +1,113 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.EventHub.Models;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using System.Collections;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Commands.EventHub.Commands.EventHub
+{
+    /// <summary>
+    /// 'New-AzEventHub' Cmdlet creates a new EventHub
+    /// </summary>
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "EventHubCluster", DefaultParameterSetName = ClusterPropertiesParameterSet, SupportsShouldProcess = true), OutputType(typeof(PSEventHubClusterAttributes))]
+    public class NewAzureRmEventHubCluster : AzureEventHubsCmdletBase
+    {
+        [Parameter(Mandatory = true, ParameterSetName = ClusterPropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+         public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = ClusterPropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "Cluster Name")]
+        [Parameter(Mandatory = true, ParameterSetName = ClusterResourceIdParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Cluster Name")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = ClusterPropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = "Location of Cluster")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        [Parameter(Mandatory = false, ParameterSetName = ClusterPropertiesParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "Cluster Capacity (CU), curerntrly, allowed value = 1")]
+        [ValidateNotNullOrEmpty]
+        public int? Capacity { get; set; }
+
+        [Parameter(Mandatory = false, ParameterSetName = ClusterPropertiesParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "Hashtables which represents resource Tags for Clusters")]
+        public Hashtable Tag { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = ClusterInputObjectParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "Cluster InputObject")]
+        [ValidateNotNullOrEmpty]
+        public PSEventHubClusterAttributes InputObject { get; set; }
```

Comment:
It is strange to create PSEventHubClusterAttributes object with input object of the type PSEventHubClusterAttributes. Does it mean a clone operation?

Reply:
will remove it, the intention was to get the properties from Input Object and Name provided in the cmdlet

Reply:
removed InputObject